x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/openai/OpenAiResponseHandler.java

```diff
@@ -61,4 +79,56 @@ void checkForFailureStatusCode(HttpRequestBase request, HttpResult result) throw
             throw new RetryException(false, buildError(UNSUCCESSFUL, request, result));
         }
     }
+
+    static ElasticsearchStatusException buildRateLimitError(String baseMessage, HttpRequestBase request, HttpResult result) {
+        var response = result.response();
+        int statusCode = result.response().getStatusLine().getStatusCode();
+        var tokenLimit = getFirstHeader(response.getHeaders(TOKENS_LIMIT));
```

Comment:
Would it make this code simpler to use `getFirstHeader()` defined in `HttpMessage` from https://hc.apache.org/httpcomponents-core-4.4.x/current/httpcore/apidocs/